### ISSUE TITLE
Typing near a link partially removes it.

### ISSUE THREAD

**Issue by chrisdeeming** [froala/wysiwyg-editor#2125]
Please see report from one of our users:

Before:
![image](https://user-images.githubusercontent.com/2158514/29036645-2f7c1f7e-7b98-11e7-81b5-6acb1ced75f2.png)

Then I press key "." or any other key on my mobile keyboard Android Chrome.

After:
SmartSelectImage_2017-06-13-08-35-00.png
![image](https://user-images.githubusercontent.com/2158514/29036651-36f977f6-7b98-11e7-96fc-d5ebd1a40c17.png)

It should not remove Link on last link text word.

I have newest Android and newest Chrome.

**Comment by chrisdeeming** [comment]
Forgot to mention: They were able to reproduce the same issue on the Froala Textarea demo.

**Comment by stefanneculai** [comment]
@chrisdeeming could you please send us the HTML you have in there? Thank you.

**Comment by mikesul** [comment]
This may be Android keyboard dependent as it looks like it's caused by word completion. Note that my initial tests have been done with SwiftKey.

I started with an empty editor, typed "test" 3 times and then highlighted the middle one and gave it a link. The resulting HTML was:

```<p>Test <a href="http://Example.com">test</a> test&nbsp;</p>```

I placed my cursor at the end of the linked "test". The link edit tooltip is displayed and my keyboard displays some autocomplete options. I choose "testing" and it removes the link. Resulting HTML:

```<p>Test testing&nbsp;&nbsp;test&nbsp;</p>```

If I redo the process but type "ing" manually, this is the resulting HTML:

```<p>Test <a href="http://Example.com">test</a>ing&nbsp;test&nbsp;</p>```

If I put my cursor in the middle of the word -- between the E and S in "test", I'm given an autocomplete option of "tests". Selecting that results in this HTML:

```<p>Test tests&nbsp;<a href="http://Example.com">st</a>&nbsp;test&nbsp;</p>```

So again, the completion has happened outside of the link (though here the text afterwards was maintained).

As a point of comparison, the behavior is a bit different using the Google keyboard (Gboard). If I have my cursor after the "test" link and click "testing", then I end up with:

```<p>Test <a href="http://Example.com">test</a>ing&nbsp;test&nbsp;</p>```

If I have it in the middle of the word and click "testing":

```<p>Test testing&nbsp;test&nbsp;</p>```

Given this, I have no idea of the viability of changes. However, at least in SwiftKey, it makes even pressing space after a link potentially remove the link.
